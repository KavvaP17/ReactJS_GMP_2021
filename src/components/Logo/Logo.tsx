import React from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
    logo: {
        color: '#f65261',
        fontSize: 24
    },
    logoBold: {
        fontWeight: 'bold'
    }
})

export const Logo = (): JSX.Element => {
    const styles = useStyles();
    return (
        <div className={styles.logo}>
            <span className={styles.logoBold}>netflix</span>
            <span>roulette</span>
        </div>
    )
}
