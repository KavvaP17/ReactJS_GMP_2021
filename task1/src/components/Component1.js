import { Component } from 'react';

export default class Component1 extends Component {
    render() {
        return (
            <p>React.Component</p>
        );
    }
}