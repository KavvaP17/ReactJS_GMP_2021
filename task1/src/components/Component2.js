import { PureComponent } from 'react';

export default class Component2 extends PureComponent {
    render() {
        return (
            <p>React.PureComponent</p>
        );
    }
}