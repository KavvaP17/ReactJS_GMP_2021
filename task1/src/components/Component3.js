const Component3 = () => {
    return (
        <p>Functional component</p>
    );
}

export default Component3;