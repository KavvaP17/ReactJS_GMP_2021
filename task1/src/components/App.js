import Component1 from './Component1';
import Component2 from './Component2';
import Component3 from './Component3';

function App() {
  return (
    <div>
      <Component1 />
      <Component2 />
      <Component3 />
    </div>
  );
}

export default App;
