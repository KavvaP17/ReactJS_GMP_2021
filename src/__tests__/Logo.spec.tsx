import React from 'react';
import { render } from '@testing-library/react';
import { Logo } from '../components/Logo';

describe('Logo', () => {
    it('renders Logo snapshot', () => {
        const {asFragment } = render( <Logo />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should contain the required text', () => {
        const {queryByText } = render( <Logo />);

        expect(queryByText(/netflix/i)).toBeTruthy();
        expect(queryByText(/roulette/i)).toBeTruthy();

    });
})
