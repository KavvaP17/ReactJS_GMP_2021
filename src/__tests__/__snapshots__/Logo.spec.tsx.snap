// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders Logo snapshot 1`] = `
<DocumentFragment>
  <div
    class="logo-0-2-1"
  >
    <span
      class="logoBold-0-2-2"
    >
      netflix
    </span>
    <span>
      roulette
    </span>
  </div>
</DocumentFragment>
`;
